File: Matlab/ConvertToNormalizedQRSComplexesWithId.m
function [ QRSComplexs ] = ConvertToNormalizedQRSComplexesWithId( raw_data , classId)

[rowsLength, columnsLength] = size(raw_data);
normalizedData = zeros(rowsLength, columnsLength);
for i = 1: rowsLength
    rawRowQRS = raw_data(i,:);
    normalizedData(i,:) = (rawRowQRS - min(rawRowQRS))/(max(rawRowQRS) - min(rawRowQRS));
end

normalizedData = normalizedData';

for i = 1 : columnsLength % i know that this is low-efficiency solution
    QRSComplexs(i) = QRSComplex([classId(i) normalizedData(i,:)]);
end
end

File: Julia/ConvertToNormalizedQRSComplexesWithId.jl
function ConvertToNormalizedQRSComplexesWithId( raw_data , classId)
    sizeOfData = size(raw_data)
    info(string("Raw data size: ", sizeOfData))
    normalizedData = zeros(sizeOfData[1], sizeOfData[2]+1)
    info(string("Normalized data size: ", size(normalizedData)))
    for i = 1: sizeOfData[1]
        rawRowQRS = raw_data[i,:];
        normalizedData[i,:] = unshift!(collect((rawRowQRS - minimum(rawRowQRS))/(maximum(rawRowQRS) - minimum(rawRowQRS))), classId[i]);
    end
    normalizedData = normalizedData'
    info(string("Normalized data size: ", size(normalizedData)))
    return normalizedData
end